.github/workflows/rust.yml

```diff
@@ -154,8 +135,8 @@ jobs:
           build-args: |
             PACKAGE=${{ env.PACKAGE }}
           context: rust/
-          cache-from: type=gha,scope=${{ env.PACKAGE }}-${{ github.sha }}
-          cache-to: type=gha,mode=max,scope=${{ env.PACKAGE }}-${{ github.sha }}
+          cache-from: type=gha,scope=rust
+          cache-to: ${{ github.ref == 'refs/heads/main' && 'type=gha,mode=max,scope=rust' }}
           file: rust/Dockerfile
           push: true
```

Comment:
Follow-up: We might want to consider to only push containers on main too.

Reply:
Or, given that we have our own CD job, maybe set this always to `false`?

Reply:
Doesn't look like the CD builds, so I updated this to only push on `main`